### ISSUE TITLE
Error not thrown inside computed property

### ISSUE THREAD

**Issue by solidevolution** [vuejs/vue-test-utils#1565]
### Version
1.0.3

### Reproduction link
[https://codesandbox.io/s/naughty-hofstadter-qsycv?file=/src/components/HelloWorld.spec.js](https://codesandbox.io/s/naughty-hofstadter-qsycv?file=/src/components/HelloWorld.spec.js)

### Steps to reproduce
1. Write a component with a computed property
3. throws an error inside the computed property
2. Try to test this error with "toThrowError"

### What is expected?
Error in computed property would be thrown

### What is actually happening?
Error in computed property wouldn't be thrown

**Comment by lmiller1990** [comment]
This could be to do with the custom error handler. https://github.com/vuejs/vue-test-utils/blob/dev/packages/test-utils/src/error.js. Vue (or VTU) could be swallowing the error.

I have tried fixing this once before but no luck.

Related: https://github.com/vuejs/vue-test-utils/issues/1503

**Comment by AtofStryker** [comment]
I actually do not think this is an issue with VTU, but instead [Jest](https://github.com/facebook/jest). I noticed in your code sandbox that an `async` function was passed into `expect`. This unfortunately doesn't work quite like most would expect it to behave. This [issue](https://github.com/facebook/jest/issues/1700) as well as this [stack overflow post](https://stackoverflow.com/questions/47144187/can-you-write-async-tests-that-expect-tothrow) should provide clarity.

In short, this will work:
```
    try {
      const wrapper = shallowMount(HelloWorld, {
        propsData: {
          color: "not-in-map"
        }
      });
      await wrapper.vm.$nextTick();
    }catch(e){
      expect(e).toEqual(new Error(`Color not found: not-in-map`))
    }
```

As well as a few other methods discussed [here](https://github.com/facebook/jest/issues/1377)

**Comment by lmiller1990** [comment]
I think you might be correct.

Might close this one, I don't think we can fix this in VTU; the above post should be a good enough work-around.

**Comment by solidevolution** [comment]
Works perfect, thx!
